var/spack/repos/builtin/packages/armpl-gcc/package.py

```diff
@@ -283,6 +298,31 @@ class ArmplGcc(Package):
             expand = extension != ".dmg"
             version(ver, sha256=sha256sum, url=url, extension=extension, expand=expand)
 
+    if spack.platforms.host().default_os.startswith("ubuntu"):
+        version(
+            "24.04",
+            sha256="a323074cd08af82f4d79988cc66088b18e47dea4b93323b1b8a0f994f769f2f0",
+            url="https://developer.arm.com/-/media/Files/downloads/hpc/arm-performance-libraries/24-04/linux/arm-performance-libraries_24.04_deb_gcc.tar",
+            extension=".tar",
+            expand=True,
+        )
+    elif spack.platforms.host().name == "darwin":
+        version(
+            "24.04",
+            sha256="228bf3a2c25dbd45c2f89c78f455ee3c7dfb25e121c20d2765138b5174e688dc",
+            url="https://developer.arm.com/-/media/Files/downloads/hpc/arm-performance-libraries/24-04/macos/arm-performance-libraries_24.04_macOS.tgz",
+            extension=".tgz",
+            expand=True,
+        )
+    else:
+        version(
+            "24.04",
+            sha256="d3917523034cf5a35e4f31f9a8bf4e53e7cc97892e89739d5757cb65ce40dc2e",
+            url="https://developer.arm.com/-/media/Files/downloads/hpc/arm-performance-libraries/24-04/linux/arm-performance-libraries_24.04_rpm_gcc.tar",
+            extension=".tar",
+            expand=True,
+        )
```

Comment:
On a second read, can we avoid conditionals based on `spack.platforms.host()` ?

Reply:
Dear @alalazo I am trying to change the code above but i'd like to test it first. 

I am trying to understand what path I have to add in `sys.path.insert` so that the `from spack.package import *` does not fail. Would you be able to advise me on how to locally make the import succeed so I can locally test rather than making a change and then locally `spack install` every time? 

Thanks very much

Reply:
I have resolved by adding:

```
sys.path.insert(0,'/home/paotri01/spack/lib/spack/external/_vendoring')
sys.path.insert(0,'/home/paotri01/spack/lib/spack/external')
sys.path.insert(0,'/home/paotri01/spack/lib/spack')
```

But please do let me know,for the future, if there is an easier way. 

Thank you very much for now.

Reply:
Hi @alalazo , should have addressed your comment. Thanks

Reply:
@tldahlgren apologies for pestering but I was wondering if there was any update on this review.

I have seen another issue installing gromacs and using armpl and, since this patch contains a code change to address the installation of gromacs, I was wondering if this patch could address that problem too.

Thank you very much for your understanding.

P

Reply:
We still need to address this change properly, but the PR is already an improvement over current state.